cloud/new_ec2_cost_test.go

```diff
@@ -275,41 +275,42 @@ func (s *CostIntegrationSuite) TestGetProviderAuto() {
 	settings := &NewEC2ProviderSettings{}
 	s.m.provider = autoProvider
 
-	fetcher := getPkgCachingPriceFetcher()
+	if pkgCachingPriceFetcher == nil {
+		pkgCachingPriceFetcher = new(cachingPriceFetcher)
+	}
+	pkgCachingPriceFetcher.Lock()
+	defer pkgCachingPriceFetcher.Unlock()
```

Comment:
would it be more reasonable to write this as a suite and not use the global instance of the fetcher?

Reply:
There's no need for locking in the test at all. I've removed it.